### PR TITLE
implemented Controllers for movie, movieSessions and auth; created DTOs

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: src/main/java/com/cinema/config/AppConfig.java [hunk 1]
```diff
@@ -16,7 +17,8 @@
 @ComponentScan(value = {
         "com.cinema.dao",
         "com.cinema.service",
-        "com.cinema.security"
+        "com.cinema.security",
+        "com.cinema.model.dto"
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/config/AppConfig.java]: Do you need dto package here?

File: src/main/java/com/cinema/controllers/MovieController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private final MovieService movieService;
+    private final ModelMapper modelMapper;
+
+    @Autowired
+    public MovieController(MovieService movieService, ModelMapper modelMapper) {
+        this.movieService = movieService;
+        this.modelMapper = modelMapper;
+    }
+
+    @GetMapping("/all")
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/controllers/MovieController.java]: I think "/all"  is unnecessary (REST Resource Naming Guide: Never use CRUD function names in URIs)

File: src/main/java/com/cinema/controllers/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public List<MovieDto> getAll() {
+        return movieService.getAll().stream()
+                .map(movie -> modelMapper.map(movie, MovieDto.class))
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/add")
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/controllers/MovieController.java]: I think "/add"  is unnecessary (REST Resource Naming Guide: Never use CRUD function names in URIs)

File: src/main/java/com/cinema/controllers/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Autowired
+    public MovieSessionController(MovieSessionService movieSessionService, ModelMapper modelMapper,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.modelMapper = modelMapper;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping("/add")
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/controllers/MovieSessionController.java]: same as in MovieController

File: src/main/java/com/cinema/controllers/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    public void addMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionMapper.mapToMovieSession(movieSessionRequestDto));
+    }
+
+    @GetMapping("/search")
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/controllers/MovieSessionController.java]:
```suggestion
    @GetMapping("/available")
```

File: src/main/java/com/cinema/controllers/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/sessions")
```
Comment (Sofasmile) [inline src/main/java/com/cinema/controllers/MovieSessionController.java]:
```suggestion
@RequestMapping("/movie-sessions")
```

File: src/main/java/com/cinema/controllers/MovieSessionController.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @PostMapping("/add")
+    public void addMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionMapper.mapToMovieSession(movieSessionRequestDto));
+    }
+
+    @GetMapping("/search")
+    public List<MovieSessionResponseDto> getAvailableSessions(@RequestParam Long movieId,
+                                                              @RequestParam String date) {
```
Comment (Sofasmile) [inline src/main/java/com/cinema/controllers/MovieSessionController.java]: let's use LocalDate here

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 8]
```diff
@@ -49,4 +49,13 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataProcessingException("Can't get all cinema halls", e);
         }
     }
+
+    @Override
+    public CinemaHall get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
```
Comment (Sofasmile) [inline src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java:55]: 1 - https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_33.html

File: src/main/java/com/cinema/model/dto/MovieDto.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class MovieDto {
```
Comment (Sofasmile) [inline src/main/java/com/cinema/model/dto/MovieDto.java]: let's use two dtos: request (without id) and response(with movie id)

File: src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class MovieSessionResponseDto {
+    private Long id;
+    private String movieTitle;
```
Comment (Sofasmile) [inline src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java:9]: let's add movie id too

File: src/main/java/com/cinema/config/AppConfig.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/config/AppConfig.java]: thanks. fixed.

File: src/main/java/com/cinema/controllers/MovieController.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/controllers/MovieController.java]: done

File: src/main/java/com/cinema/controllers/MovieController.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/controllers/MovieController.java]: done

File: src/main/java/com/cinema/controllers/MovieSessionController.java (hunk 4 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/cinema/controllers/MovieSessionController.java (hunk 5 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/cinema/controllers/MovieSessionController.java (hunk 6 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/cinema/controllers/MovieSessionController.java (hunk 7 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 8 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java:55]: done

File: src/main/java/com/cinema/model/dto/MovieDto.java (hunk 9 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/model/dto/MovieDto.java]: done

File: src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java (hunk 10 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java:9]: done

File: src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java [hunk 11]
```diff
@@ -5,6 +5,7 @@
 @Data
 public class MovieSessionResponseDto {
     private Long id;
+    private Long movieId;
```
Comment (Sofasmile) [inline src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java:8]: let's set this field in the mapper)

Comment (TimoxaV) [reply src/main/java/com/cinema/model/dto/MovieSessionResponseDto.java:8]: I use ModelMapper to convert MovieSession to MovieSessionResponseDto. That's why I don't set it in the mapper.

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**